### ISSUE TITLE
Changing explorer's pop-up dialogs titlebar while keeping navigation pane default look

### ISSUE THREAD

**Issue by UknownGino** [MicaForEveryone/MicaForEveryone#108]
By default MFE sets explorer backdrop to default with a custom rule. By doing so, all the pop-up dialogs such as Drive/File properties and explorer's options use the default white titlebar without any backdrop applied.

Changing the backdrop type of explorer also changes the titlepane look of the main window.
Is there any way to only apply a different backdrop on dialogs like properties etc...?

### Explorer and Dialogs with default backdrop applied
![image](https://user-images.githubusercontent.com/61624086/182859785-62045f8f-8d4f-4f51-8c2c-2bf5152b3d70.png)

### Explorer and Dialogs with Acrylic backdrop applied
![image](https://user-images.githubusercontent.com/61624086/182861437-da1a723c-fc45-4fb8-b157-d69ab780a092.png)

### What i want to achieve
![image](https://user-images.githubusercontent.com/61624086/182861285-d373d441-b227-408d-b2ea-c6a53ad0b208.png)

**Comment by xmine64** [comment]
Find Properties' window class name and set rule using its class name.

**Comment by UknownGino** [comment]
Ok, I've found it using Spy++ and seems to be #32770, wich seems to be present in Class Rule, but even after adding the rule the window stays the same (obviously I've already tried re-opening it)

![image](https://user-images.githubusercontent.com/61624086/182879820-b3d61077-b559-4206-9881-cecfaf34513e.png)

**Comment by dongle-the-gadget** [comment]
> Ok, I've found it using Spy++ and seems to be #32770, wich seems to be present in Class Rule, but even after adding the rule the window stays the same (obviously I've already tried re-opening it)
> 
> ![image](https://user-images.githubusercontent.com/61624086/182879820-b3d61077-b559-4206-9881-cecfaf34513e.png)

Uh can you show us the Spy++ window?

**Comment by UknownGino** [comment]
Sure, here you are:

![image](https://user-images.githubusercontent.com/61624086/182880808-a5cd4f3d-753f-4e94-9be2-93654ce43b0a.png)

On MFE the class name is only "#32770" and doesnt have "(Dialog)" appended. But I don't think thats the issue.

**Comment by UknownGino** [comment]
Nothing, same thing :(

![image](https://user-images.githubusercontent.com/61624086/182882314-dcbd025b-2467-4821-8b84-5c0c7fd512de.png)

And it only suggests the one without:

![image](https://user-images.githubusercontent.com/61624086/182882465-7c983c47-85f5-435d-9ede-fd2e66a0b3b4.png)

**Comment by dongle-the-gadget** [comment]
OK, so I figured out the problem here. You need to move the class rule to before explorer.exe, since MFE checks for rules from top to bottom, so if the \#32770 rule is lower than the explorer.exe rule, the explorer.exe rule takes presence.

**Comment by dongle-the-gadget** [comment]
Technically not a bug, just confusing app design. @xmine64.

**Comment by xmine64** [comment]
> Technically not a bug, just confusing app design.

Well I didn't think about this, but we can give higher priority to class rules.

**Comment by UknownGino** [comment]
Ok thank you, that actually works! I've moved the class up by editing the conf file.

@xmine64 it would be pretty cool if you could add "up" and "down" buttons to edit the priority of the rule directly via app. Should I open another issue or leave this one open to suggest that?

**Comment by dongle-the-gadget** [comment]
> > Technically not a bug, just confusing app design.
> 
> Well I didn't think about this, but we can give higher priority to class rules.

Or have the add rule buttons add the rules to right below the global rules?

**Comment by xmine64** [comment]
> it would be pretty cool if you could add "up" and "down" buttons to edit the priority of the rule directly via app. Should I open another issue or leave this one open to suggest that?

Yes, we need those too. I guess editing issue's title is enough.

**Comment by UknownGino** [comment]
Oh and it would be cool to if rules could have custom displaynames as some window classes don't have proper names (like this one) and if in higher amounts can be difficult to rembember what is what

**Comment by xmine64** [comment]
> it would be cool to if rules could have custom displaynames

Well you can add comments in your config file, but there's nothing about it in GUI. Also class rules are not specific to a process, that means if another process create a class with "#32770" name then this rule will be applied to that window as well. I guess we need to add support for rules that specify class and rule togother.

**Comment by UknownGino** [comment]
I don't know if this is also a problem with rules priority but while running MFE and having Acrylic applied breaks Windows Terminal.

The Terminal window renders properly but there are some artifacts on the desktop. Obviously Windows Terminal doesn't need MFE to have titlebar effects, so I've tried to disable it.

This time i've applied rules for both the window class (`Windows.UI.Composition.DesktopWindowContentBridge`) and for the executable (`WindowsTerminal.exe`) but with no results.

### Spy++ window inspection
![image](https://user-images.githubusercontent.com/61624086/183252629-e0716316-1f91-434b-bd56-71c8b543aebe.png)

### This is what happens when Terminal is not minimized
![image](https://user-images.githubusercontent.com/61624086/183252517-aa070034-af88-412e-9dd0-13868a0a7fec.png)
_Square appears on top left desktop corner_

### This is what happens when Terminal is minimized
![image](https://user-images.githubusercontent.com/61624086/183252556-35a4e53f-ebb3-428f-b54e-d1b3f4a1500f.png)
_Rectangle appears on bottom left desktop corner_

**Comment by dongle-the-gadget** [comment]
@UknownGino the window you selected isn't actually Windows Terminal, but rather, the XAML Islands child window that Windows Terminal has.

**Comment by UknownGino** [comment]
Seems that I am not able to select the proper window then, everywhere I point with the finder tool in spy++ always has the same window class. :(

**Comment by dongle-the-gadget** [comment]
The window class is CASCADIA_HOSTING_WINDOW_CLASS btw.

**Comment by UknownGino** [comment]
I've added it as the first rule, but it still renders the extra rectangle and square.

![image](https://user-images.githubusercontent.com/61624086/183260894-3fd0b0a7-a033-46d9-a9c6-63ebcbed8828.png)

**Comment by UknownGino** [comment]
> I've added it as the first rule, but it still renders the extra rectangle and square.

Any suggestions on how to fix it?

**Comment by dongle-the-gadget** [comment]
@UknownGino, you might want to check out 9a8d61f.

**Comment by UknownGino** [comment]
@dongle-the-gadget Thank you, that fixes the terminal issue :)
